java/connector-node/risingwave-connector-service/src/main/java/com/risingwave/connector/source/core/JniDbzSourceHandler.java

```diff
@@ -0,0 +1,132 @@
+        var config =
+                new DbzConnectorConfig(
+                        SourceTypeE.valueOf(request.getSourceType()),
+                        request.getSourceId(),
+                        request.getStartOffset(),
+                        mutableUserProps,
+                        request.getSnapshotDone());
+        JniDbzSourceHandler handler = new JniDbzSourceHandler(config);
+        handler.start(channelPtr);
+    }
+
+    class OnReadyHandler implements Runnable {
+        private final CdcEngineRunner runner;
+        private final long channelPtr;
+
+        public OnReadyHandler(CdcEngineRunner runner, long channelPtr) {
+            this.runner = runner;
+            this.channelPtr = channelPtr;
+        }
+
+        @Override
+        public void run() {
+            while (runner.isRunning()) {
+                try {
+                    // check whether the send queue has room for new messages
+                    // Thread will block on the channel to get output from engine
+                    var resp =
+                            runner.getEngine().getOutputChannel().poll(500, TimeUnit.MILLISECONDS);
+                    boolean success;
+                    if (resp != null) {
+                        ConnectorNodeMetrics.incSourceRowsReceived(
+                                config.getSourceType().toString(),
+                                String.valueOf(config.getSourceId()),
+                                resp.getEventsCount());
+                        LOG.info(
+                                "Engine#{}: emit one chunk {} events to network ",
+                                config.getSourceId(),
+                                resp.getEventsCount());
+                        success = Binding.sendCdcSourceMsgToChannel(channelPtr, resp.toByteArray());
+                    } else {
+                        // If resp is null means just check whether channel is closed.
+                        success = Binding.sendCdcSourceMsgToChannel(channelPtr, null);
+                    }
+                    if (!success) {
+                        LOG.info(
+                                "Engine#{}: JNI sender broken detected, stop the engine",
+                                config.getSourceId());
+                        runner.stop();
+                        return;
+                    }
+                } catch (Throwable e) {
+                    LOG.error("Poll engine output channel fail. ", e);
+                }
+            }
+        }
+    }
+
+    public void start(long channelPtr) {
+        var runner = DbzCdcEngineRunner.newCdcEngineRunner(config);
+        if (runner == null) {
+            return;
+        }
+
+        try {
+            // Start the engine
+            runner.start();
+            LOG.info("Start consuming events of table {}", config.getSourceId());
+
+            final OnReadyHandler onReadyHandler = new OnReadyHandler(runner, channelPtr);
```

Comment:
Since we are switching the channel to a bounded rust channel, we can get rid of the concept of `ReadyHandler`. The `ReadyHandler` is for [flow control of grpc](https://github.com/grpc/grpc-java/issues/2247#issuecomment-245949881), actually it is a callback provided to the grpc framework. That is it will **alive** after the previous `startSource()` function returned. 
And it seems that the `ReadyHandler` becomes a main loop of a worker thread and the Rust side will spawn a thread for it indeed.